src/main/java/Entities/Item.java

```diff
@@ -179,21 +178,17 @@ public void setPriceHistoryData(ArrayList<Double> updatedPrices){
     public int getReviewCount() { return reviewCount;}
     public void setProductPrice(double newPrice) {this.itemPrice = newPrice;}
     public void setProductCurrency(String newCurrency) {this.itemCurrency = newCurrency;}
+    public void setDateLastUpdated(Date date) {this.dateLastUpdated = date;}
+
 
     /** Updates price of Item object through web-scraping the product page on Amazon
      * */
     public void updatePrice() throws IOException{
-        try {
-            // This line specifies window type and layout of amazon page based on  Window Version and browser for webscraping
-            Document doc = Jsoup.connect(url).timeout(10000).userAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36").get();
-            Element price = doc.select(".a-offscreen").first();
-            assert price != null;
-            double sellingPrice = Double.parseDouble(price.text().substring(1));
-            priceChange = itemPrice - sellingPrice;
-            itemPrice = sellingPrice;
-            dateLastUpdated = new Date();
-        } catch (IOException e){
-            e.printStackTrace();
-        }
+        ItemUpdateChecker itemUpdateChecker = new ItemUpdateChecker();
+        double sellingPrice = itemUpdateChecker.updatePriceCheck(url);
```

Comment:
This refactored function is still not following clean architecture as it violates the dependency tree of an external interface being used inside of an Entity. The proper way to achieve clean architecture would be to move all of this logic into the external interface and pass a Product into that instead of here and using the products setters to achieve the price updates

Reply:
Hello, thanks for the feedback. I have completely isolated the update price check function from the item class to follow clean architecture. I have added it to a newly defined external interface class called "ItemUpdateChecker".